openassessment/xblock/staff_area_mixin.py

```diff
@@ -370,8 +364,24 @@ def get_student_info_path_and_context(self, student_username, expanded_view=None
             submission = submissions[0]
             submission_uuid = submission['uuid']
 
+        # This will add submission (which may be None) and username to the context.
         context = self.get_student_submission_context(student_username, submission)
 
+        # Only add the rest of the details to the context if a submission exists.
+        if submission_uuid:
+            self.add_submission_context(submission_uuid, context)
+
+        path = 'openassessmentblock/staff_area/oa_student_info.html'
+        return path, context
+
+    def add_submission_context(self, submission_uuid, context):
```

Comment:
I broke this out into its own method just to make get_student_info_path_and_context more readable.